src/ray/core_worker/core_worker.cc

```diff
@@ -1174,6 +1174,13 @@ void CoreWorker::Exit(
     const rpc::WorkerExitType exit_type,
     const std::string &detail,
     const std::shared_ptr<LocalMemoryBuffer> &creation_task_exception_pb_bytes) {
+  if (is_exit_) {
+    RAY_LOG(INFO) << "Exit signal received, but the core worker has already received a "
+                     "signal and is shutting down.";
+    return;
+  }
+  is_exit_ = true;
```

Comment:
(not familiar with the code, just question for my curiosity)
is it possible multiple threads checking `is_exit_` and all of them pass through?
If it's not supposed to be called in multi-threaded env, we don't need atomic variable here.

In a word, reading through the code, I think you should use CAS
Ref: https://en.cppreference.com/w/cpp/atomic/atomic/compare_exchange